src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/flow/FlowAnnotatorBase.java

```diff
@@ -424,19 +441,24 @@ protected void cycleSkip(final VariantContext vc, final LocalContext localContex
         localContext.attributes.put(GATKVCFConstants.FLOW_CYCLESKIP_STATUS, css);
     }
 
-    // get a single nucleoid from reference
+    // get a single nucleotid from reference
     private byte getReferenceNucleotide(final LocalContext localContext, final int start) {
         final int         index = start - localContext.ref.getWindow().getStart();
         final byte[]      bases = localContext.ref.getBases();
-        Utils.validIndex(index, bases.length);
+        Utils.validIndex(index, bases.length); // do not catch, if here the location is outside of the reference, there is a problem!
         return bases[index];
     }
 
     // get an hmer from reference plus a number of additional bases
     private byte[] getReferenceHmerPlus(final LocalContext localContext, final int start, final int additional) {
         int               index = start - localContext.ref.getWindow().getStart();
         final byte[]      bases = localContext.ref.getBases();
-        Utils.validIndex(index, bases.length);
+        try {
+            Utils.validIndex(index, bases.length);
+        } catch (IllegalArgumentException e) {
+            flowMissingOneShotLogger.warn("Failed to get hmer from reference. Start index: " + index + " Reference length: " + bases.length);
```

Comment:
Is this also tied to being toward the end of the reference? You aren't asking the reference context for its actual length but the lenght of the bases view which is arbitrary and going to be deeply confusing to users. This error message should either try to give the actual reference name and index or give a clue as to why this could happen since that bases.length view is confusing.

Reply:
Fixed, could not find an easy way to get the length of the reference, `localContext.ref.getContigLength` is somehow hidden